File: frontend/src/types/log.ts
export interface LogEntry {
  logId: string
  operatorId: string
  operatorName: string
  operationType: string
  operationTypeLabel: string
  targetType: string
  targetId: string
  targetName: string
  detail: string
  ip: string
  level: string
  createdAt: string
  fieldChanges?: FieldChange[]
}

export interface FieldChange {
  fieldName: string
  fieldLabel: string
  oldValue: string | null
  newValue: string | null
}

export interface LogQuery {
  page: number
  pageSize: number
  keyword?: string
  level?: string[]
  operation?: string
  module?: string
  startTime?: string
  endTime?: string
  username?: string
}

export const LOG_LEVELS = [
  { value: 'info', label: 'INFO' },
  { value: 'warn', label: 'WARN' },
  { value: 'error', label: 'ERROR' },
  { value: 'debug', label: 'DEBUG' },
]

export const LOG_LEVEL_CONFIG: Record<string, { color: string; label: string }> = {
  info: { color: '#3b82f6', label: 'INFO' },
  warn: { color: '#f59e0b', label: 'WARN' },
  error: { color: '#ef4444', label: 'ERROR' },
  debug: { color: '#64748b', label: 'DEBUG' },
}

export const METHOD_COLORS: Record<string, string> = {
  GET: '#10b981',
  POST: '#3b82f6',
  PUT: '#f59e0b',
  DELETE: '#ef4444',
  PATCH: '#8b5cf6',
}

File: backend/src/main/java/com/sms/nexus/entity/Log.java
package com.sms.nexus.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("logs")
public class Log {

    @TableId("log_id")
    private String logId;

    @TableField("operator_id")
    private String operatorId;

    @TableField("operator_name")
    private String operatorName;

    @TableField("operation_type")
    private String operationType;

    @TableField("target_type")
    private String targetType;

    @TableField("target_id")
    private String targetId;

    @TableField("target_name")
    private String targetName;

    @TableField("detail")
    private String detail;

    @TableField("ip")
    private String ip;

    @TableField("level")
    private String level;

    @TableField("created_at")
    private LocalDateTime createdAt;
}
